From 93b94b9579c475fe212cbb58919f3da2158640a8 Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Fri, 8 Jun 2018 21:13:00 +0800
Subject: [PATCH] arm: dts: rockchip: add saradc keys for px3se evb

The px3se evb board has a adc keyboard including Vol+, Vol-,
ESC, Home, Menu, Reserve.

Change-Id: Id3794d7a845f7c5364900865440a731d162a9ad0
Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>
---
 arch/arm/boot/dts/px3se-evb.dts | 50 +++++++++++++++++++++++++++++++++
 1 file changed, 50 insertions(+)

diff --git a/arch/arm/boot/dts/px3se-evb.dts b/arch/arm/boot/dts/px3se-evb.dts
index dc6eb3b67012..120481842301 100644
--- a/arch/arm/boot/dts/px3se-evb.dts
+++ b/arch/arm/boot/dts/px3se-evb.dts
@@ -3,6 +3,7 @@
 
 #include "rk3128.dtsi"
 #include <dt-bindings/display/drm_mipi_dsi.h>
+#include <dt-bindings/input/input.h>
 
 / {
 	model = "Rockchip PX3SE Evaluation Board";
@@ -12,6 +13,50 @@
 		bootargs = "loglevel=7 root=PARTUUID=614e0000-0000-4b53-8000-1d28000054a9 rootwait";
 	};
 
+	adc-keys {
+		compatible = "adc-keys";
+		io-channels = <&saradc 1>;
+		io-channel-names = "buttons";
+		poll-interval = <100>;
+		keyup-threshold-microvolt = <1800000>;
+
+		reserved-key {
+			linux,code = <KEY_RESERVED>;
+			label = "reserved";
+			press-threshold-microvolt = <1549000>;
+		};
+
+		home-key {
+			linux,code = <KEY_HOME>;
+			label = "home";
+			press-threshold-microvolt = <1314000>;
+		};
+
+		esc-key {
+			linux,code = <KEY_ESC>;
+			label = "esc";
+			press-threshold-microvolt = <985000>;
+		};
+
+		menu-key {
+			linux,code = <KEY_MENU>;
+			label = "menu";
+			press-threshold-microvolt = <623000>;
+		};
+
+		vol-down-key {
+			linux,code = <KEY_VOLUMEDOWN>;
+			label = "volume down";
+			press-threshold-microvolt = <300000>;
+		};
+
+		vol-up-key {
+			linux,code = <KEY_VOLUMEUP>;
+			label = "volume up";
+			press-threshold-microvolt = <18000>;
+		};
+	};
+
 	backlight: backlight {
 		compatible = "pwm-backlight";
 
@@ -418,6 +463,11 @@
 	status = "okay";
 };
 
+&saradc {
+	status = "okay";
+	vref-supply = <&vcc_io>;
+};
+
 &sdio {
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
-- 
2.35.3

